### ISSUE TITLE
[Bug]: Hang on Create/Load Game

### ISSUE THREAD

**Issue by Sweek9** [cech12/UnlitCampfire#35]
### Minecraft Version

1.20.6

### Mod Loader

Forge

### Mod Loader Version

50.0.13

### Mod Version

2.4.0.1

### Describe the Issue

```
---- Minecraft Crash Report ----
// Why is it breaking :(

Time: 2024-05-13 15:11:01
Description: Exception in server tick loop

java.lang.IllegalArgumentException: Cannot set property BooleanProperty{name=infinite, clazz=class java.lang.Boolean, values=[true, false]} as it does not exist in Block{minecraft:campfire}
	at TRANSFORMER/minecraft@1.20.6/net.minecraft.world.level.block.state.StateHolder.setValue(StateHolder.java:121) ~[forge-1.20.6-50.0.13-client.jar!/:?] {re:classloading}
	at TRANSFORMER/unlitcampfire@1.20.6-2.4.0.1/de.cech12.unlitcampfire.platform.ForgeConfigHelper.setGeneratedCampfireIsLitInfinitely(ForgeConfigHelper.java:207) ~[unlitcampfire-forge-1.20.6-2.4.0.1.jar!/:1.20.6-2.4.0.1] {re:classloading}
	at TRANSFORMER/unlitcampfire@1.20.6-2.4.0.1/de.cech12.unlitcampfire.platform.ForgeConfigHelper.loadChangedConfigData(ForgeConfigHelper.java:215) ~[unlitcampfire-forge-1.20.6-2.4.0.1.jar!/:1.20.6-2.4.0.1] {re:classloading}
	at TRANSFORMER/unlitcampfire@1.20.6-2.4.0.1/de.cech12.unlitcampfire.platform.ForgeConfigHelper.onLoad(ForgeConfigHelper.java:222) ~[unlitcampfire-forge-1.20.6-2.4.0.1.jar!/:1.20.6-2.4.0.1] {re:classloading}
	at TRANSFORMER/unlitcampfire@1.20.6-2.4.0.1/de.cech12.unlitcampfire.platform.__ForgeConfigHelper_onLoad_Loading.invoke(.dynamic) ~[unlitcampfire-forge-1.20.6-2.4.0.1.jar!/:1.20.6-2.4.0.1] {re:classloading,pl:eventbus:B}
	at SECURE-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:48) ~[eventbus-6.2.6.jar!/:?] {}
	at SECURE-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:304) ~[eventbus-6.2.6.jar!/:?] {}
	at SECURE-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:290) ~[eventbus-6.2.6.jar!/:?] {}
	at LAYER PLUGIN/net.minecraftforge.javafmlmod@1.20.6-50.0.13/net.minecraftforge.fml.javafmlmod.FMLModContainer.lambda$new$0(FMLModContainer.java:42) ~[javafmllanguage-1.20.6-50.0.13.jar!/:50.0.13] {}
	at LAYER PLUGIN/net.minecraftforge.fmlcore@1.20.6-50.0.13/net.minecraftforge.fml.ModContainer.lambda$dispatchConfigEvent$9(ModContainer.java:171) ~[fmlcore-1.20.6-50.0.13.jar!/:1.0] {}
	at java.base/java.util.Optional.ifPresent(Optional.java:178) ~[?:?] {}
	at LAYER PLUGIN/net.minecraftforge.fmlcore@1.20.6-50.0.13/net.minecraftforge.fml.ModContainer.dispatchConfigEvent(ModContainer.java:171) ~[fmlcore-1.20.6-50.0.13.jar!/:1.0] {}
	at LAYER PLUGIN/net.minecraftforge.fmlcore@1.20.6-50.0.13/net.minecraftforge.fml.config.ModConfig.fireEvent(ModConfig.java:77) ~[fmlcore-1.20.6-50.0.13.jar!/:?] {}
	at LAYER PLUGIN/net.minecraftforge.fmlcore@1.20.6-50.0.13/net.minecraftforge.fml.config.ConfigTracker.openConfig(ConfigTracker.java:62) ~[fmlcore-1.20.6-50.0.13.jar!/:?] {}
	at LAYER PLUGIN/net.minecraftforge.fmlcore@1.20.6-50.0.13/net.minecraftforge.fml.config.ConfigTracker.lambda$loadConfigs$1(ConfigTracker.java:50) ~[fmlcore-1.20.6-50.0.13.jar!/:?] {}
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?] {}
	at java.base/java.util.Collections$SynchronizedCollection.forEach(Collections.java:2351) ~[?:?] {}
	at LAYER PLUGIN/net.minecraftforge.fmlcore@1.20.6-50.0.13/net.minecraftforge.fml.config.ConfigTracker.loadConfigs(ConfigTracker.java:50) ~[fmlcore-1.20.6-50.0.13.jar!/:?] {}
	at TRANSFORMER/forge@50.0.13/net.minecraftforge.server.ServerLifecycleHooks.handleServerAboutToStart(ServerLifecycleHooks.java:79) ~[forge-1.20.6-50.0.13-universal.jar!/:?] {re:classloading}
	at TRANSFORMER/minecraft@1.20.6/net.minecraft.client.server.IntegratedServer.initServer(IntegratedServer.java:75) ~[forge-1.20.6-50.0.13-client.jar!/:?] {re:classloading,pl:runtimedistcleaner:A}
	at TRANSFORMER/minecraft@1.20.6/net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:659) ~[forge-1.20.6-50.0.13-client.jar!/:?] {re:classloading,pl:accesstransformer:B}
	at TRANSFORMER/minecraft@1.20.6/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:261) ~[forge-1.20.6-50.0.13-client.jar!/:?] {re:classloading,pl:accesstransformer:B}
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.20.6
	Minecraft Version ID: 1.20.6
	Operating System: Mac OS X (aarch64) version 14.4.1
	Java Version: 21.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 1394606080 bytes (1330 MiB) / 1858076672 bytes (1772 MiB) up to 34359738368 bytes (32768 MiB)
	CPUs: 20
	Processor Vendor: Apple Inc.
	Processor Name: Apple M1 Ultra
	Identifier: Apple Inc. Family 0x1b588bb3 Model 0 Stepping 0
	Microarchitecture: ARM64 SoC: Firestorm + Icestorm
	Frequency (GHz): 3.23
	Number of physical packages: 1
	Number of physical CPUs: 20
	Number of logical CPUs: 20
	Graphics card #0 name: Apple M1 Ultra
	Graphics card #0 vendor: Apple (0x106b)
	Graphics card #0 VRAM (MB): 0.00
	Graphics card #0 deviceId: unknown
	Graphics card #0 versionInfo: unknown
	Memory slot #0 capacity (MB): 0.00
	Memory slot #0 clockSpeed (GHz): 0.00
	Memory slot #0 type: unknown
	Virtual memory max (MB): 65536.00
	Virtual memory used (MB): 30625.45
	Swap memory total (MB): 0.00
	Swap memory used (MB): 0.00
	JVM Flags: 11 total; -XX:+UnlockExperimentalVMOptions -XX:+UseZGC -XX:-ZUncommit -XX:ZCollectionInterval=5 -XX:ZAllocationSpikeTolerance=2.0 -XX:+AlwaysPreTouch -XX:+ParallelRefProcEnabled -XX:+DisableExplicitGC -Xms1513M -Xmx32768M -Xss2M
	Server Running: true
	Player Count: 0 / 8; []
	Active Data Packs: vanilla, mod:forge (incompatible), mod:unlitcampfire (incompatible)
	Available Data Packs: bundle, mod:forge (incompatible), mod:unlitcampfire (incompatible), trade_rebalance, update_1_21, vanilla
	Enabled Feature Flags: minecraft:vanilla
	World Generation: Stable
	World Seed: -1272651210010856806
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'forge'; Server brand changed to 'forge'
	Launched Version: 1.20.6-forge-50.0.13
	ModLauncher: 10.2.1
	ModLauncher launch target: forge_client
	ModLauncher naming: mcp
	ModLauncher services: 
		/ slf4jfixer PLUGINSERVICE 
		/ runtimedistcleaner PLUGINSERVICE 
		/ runtime_enum_extender PLUGINSERVICE 
		/ object_holder_definalize PLUGINSERVICE 
		/ capability_token_subclass PLUGINSERVICE 
		/ accesstransformer PLUGINSERVICE 
		/ eventbus PLUGINSERVICE 
		/ mixin PLUGINSERVICE 
		/ fml TRANSFORMATIONSERVICE 
		/ mixin TRANSFORMATIONSERVICE 
	FML Language Providers: 
		lowcodefml@50
		minecraft@1.0
		javafml@50.0.13
	Mod List: 
		forge-1.20.6-50.0.13-client.jar                   |Minecraft                     |minecraft                     |1.20.6              |DONE      |Manifest: NOSIGNATURE
		forge-1.20.6-50.0.13-universal.jar                |Forge                         |forge                         |50.0.13             |DONE      |Manifest: NOSIGNATURE
		unlitcampfire-forge-1.20.6-2.4.0.1.jar            |Unlit Campfire                |unlitcampfire                 |1.20.6-2.4.0.1      |DONE      |Manifest: NOSIGNATURE
	Crash Report UUID: afe3edfe-5419-4483-854d-402323d43f13
	FML: 0.0
	Forge: net.minecraftforge:50.0.13
```

**Comment by cech12** [comment]
Thanks for your report! This is a known issue and is listed in the changelog :) But I will leave this issue open until it is fixed :)

**Comment by cech12** [comment]
Fixed in upcoming mod version 1.20.6-2.4.0.2
